trezor/trezor-suite
# Process modes & button requests refactoring

## Description

Optional followup of #7952. Mostly refactoring. I'm not saying that all of it is a good idea, let's discuss in case of doubt.

## Commits

- https://github.com/trezor/trezor-suite/pull/7964/commits/caaab6e743a01cfeadc2f3e52e3722d4f2a0c12d - Device process modes removed. Their only function was to keep modals opened by blocking `ui-close_window` event in `actionBlockerMiddleware`, which was a bit too heavyweight. Replaced by `@modal/preserve` action which simply disallow to close modal by `ui-close_window` event until explicitly closed.
- https://github.com/trezor/trezor-suite/pull/7964/commits/4f81651bd7fd3192a47fd740f720088d6ce31177 - `<ConfirmAddress />` modal is now opened directly by `ButtonRequest_Address` event (in the same way as `<ReviewTransaction />` modal, for example), which allows to remove explicit subscriptions to button requests from React components.
- https://github.com/trezor/trezor-suite/pull/7964/commits/fd7d96788c728f34286ea1cf60e5a11ddf0a7896 - necessary changes in unit tests
- https://github.com/trezor/trezor-suite/pull/7964/commits/420441864314fe48d662b6735ccf2ee4c166f59a - Cardano signing bug needed to be fixed again after removing process modes. Now `ButtonRequest_Other` is replaced by `ButtonRequest_SignTx` in `buttonRequestMiddleware` whenever `cardano` account is opened on `wallet-send` route. Still not very nice though.

## QA
Please test that with both old and new TT firmware
- every show/verify address action works correctly (confirmed or cancelled), in
  - receive address flow
  - sign & verify form
  - coinmarket (don't know much about it)
- send transaction signing process works correctly (modals without any flashing)
  - for any coin
  - and especially for Cardano
